### PR TITLE
Fixes with math classes

### PR DESCRIPTION
**Pull request by MateuszKlatecki**
- remove unnecessary #if (DP_FLOATINGPOINT == TRUE) conditions in R4 (float) functions
- rework Math.Internal class

## Motivation and Context

- After this change its possible to use double precision and single precision operations

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Signed-off-by: Mateusz Klatecki <mateusz.klatecki@gc5.pl>

### CODE REVIEW

File: src/CLR/System.Math/nf_native_system_math_System_Math.cpp [hunk 1]
```diff
@@ -1139,9 +967,9 @@
 
         double d = stack.Arg0().NumericByRefConst().r8;
         double res = 0.0;
-        double retVal = System::Math::Truncate(d, res);
```
Comment (josesimoes) [inline src/CLR/System.Math/nf_native_system_math_System_Math.cpp]: Nice catch here!
Can you please make this into a separate PR? Need to keep changes atomic... and this one it's clearly a bug fix...

File: src/CLR/CorLib/corlib_native_System_MathInternal.cpp [hunk 2]
```diff
@@ -5,99 +5,44 @@
 //
 #include "CorLib.h"
 
-#if !defined(NANOCLR_EMULATED_FLOATINGPOINT)
```
[the remark this answers is not in the document]
Comment (MateuszKlatecki) [reply src/CLR/CorLib/corlib_native_System_MathInternal.cpp]: @josesimoes but MathInternal class has nothing to do with floating points. There are only 3 functions operating on ints.
In addition, this class does not compile if DP_FLOATINGPOINT is enabled in the build options

File: src/CLR/System.Math/nf_native_system_math_System_Math.cpp (hunk 1 above)
Comment (MateuszKlatecki) [reply src/CLR/System.Math/nf_native_system_math_System_Math.cpp]: I can make this into separate PR but without this change file does not compile if DP_FLOATINGPOINT is enabled in the build options

Comment (josesimoes) [reply src/CLR/System.Math/nf_native_system_math_System_Math.cpp]: Please do that. As we don't have any target building with DP FP this hasn't been catch....

File: src/CLR/CorLib/corlib_native_System_MathInternal.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (josesimoes) [reply src/CLR/CorLib/corlib_native_System_MathInternal.cpp]: You are correct. There were those calls to FP math lib that aren't there anymore.

File: src/CLR/System.Math/nf_native_system_math_System_Math.cpp (hunk 1 above)
Comment (MateuszKlatecki) [reply src/CLR/System.Math/nf_native_system_math_System_Math.cpp]: I moved this change into separate PR
